File: python/staircase.py
from __future__ import print_function

def staircase(n):
  for i in range(0,n):
    j = n
    while j > 0:
      if (j <= i+1):
        print("x", end="")
      else:
        print(" ",end="")
      j -= 1
    print("")
    
  

print(staircase(25))

File: java/staircase.java
public class Main {
    static void staircase(int n) {
      for (int i = 0; i < n; i++) {
        for (int j = n; j > 0; j--) {                    
          if (j <= i+1) {
            System.out.print("#");
          } else {
            System.out.print(" ");
          }
        }
        System.out.println();
      }
    }

    public static void main(String[] args) {
      staircase(25);
    }
}